### PR TITLE
[12.0][FIX] mass_mailing_partner: partner merge

### PR DESCRIPTION
**Pull request by victoralmau**
Solved error related to merge contacts (in update records related to partners, exists multiple mailing contact related to same partner).

**Reproduce error**
Create partner A > Add to mailing list A
Create partner B > Add to mailing list A + B
Merge partner A + B into A

**Expected result**:
Record related to Partner A > Mailing list A + B

Result (with errors):
Record related to Partner A > Mailing list A
Record related to Partner A > Mailing list A + B

@Tecnativa TT26380

### CODE REVIEW

File: mass_mailing_partner/wizard/partner_merge.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                if not item_contact_id:
+                    self.env['mail.mass_mailing.contact'].create({
+                        'partner_id': dst_partner.id
+                    })
+                    item_contact_id = self.mass_mailing_contact_items(dst_partner.id)
+                # remove all records related to partner_ids
+                contact_ids = self.env['mail.mass_mailing.contact'].search([
+                    ('partner_id', 'in', partner_ids),
+                    ('partner_id', '!=', dst_partner.id)
+                ])
```
Comment (joao-p-marques) [inline mass_mailing_partner/wizard/partner_merge.py]: I just have a small doubt: Is this going to conflict with the deletion done by the `super()` method?

**Review by joao-p-marques — APPROVED** [review]
Just a small comment... Everything LGTM :+1:

File: mass_mailing_partner/wizard/partner_merge.py (hunk 1 above)
Comment (victoralmau) [reply mass_mailing_partner/wizard/partner_merge.py]: IMO not because search before to assign `list_ids` and remove "unnecessary" records.

Comment (joao-p-marques) [reply mass_mailing_partner/wizard/partner_merge.py]: Ok, it shouldn't be a problem, indeed, just wanted to make sure! :+1:

**Comment by victoralmau** [comment]
Can review it @chienandalu  please?

File: mass_mailing_partner/wizard/partner_merge.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                if contact_ids:
+                    list_ids = all_contact_ids.mapped('list_ids').ids
+                    list_ids = (
+                        list_ids + item_contact_id.list_ids.ids
+                    )
+                    contact_ids.unlink()
```
Comment (pedrobaeza) [inline mass_mailing_partner/wizard/partner_merge.py]: You are missing here some info, like unsubscribe information.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: mass_mailing_partner/wizard/partner_merge.py (hunk 2 above)
Comment (victoralmau) [reply mass_mailing_partner/wizard/partner_merge.py]: Thanks, in previous contact search filter to items with opt_out=False

File: mass_mailing_partner/wizard/partner_merge.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        if dst_partner:
+            contact_ids = self.env['mail.mass_mailing.contact'].search([
+                ('partner_id', 'in', partner_ids)
+            ])
+            if contact_ids:
+                contact_ids.sorted(lambda x: 1 if x.partner_id == dst_partner else 0)
+                contact_ids[0].partner_id = dst_partner
+                contact_ids[0].list_ids = [[6, 0, contact_ids.mapped('list_ids').ids]]
```
Comment (pedrobaeza) [inline mass_mailing_partner/wizard/partner_merge.py]: If not, you are removing other possible linked list

```suggestion
                contact_ids[0].list_ids = [(4, x) for x in contact_ids.mapped('list_ids').ids]]
```

File: mass_mailing_partner/wizard/partner_merge.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                contact_ids[1:].unlink()
+
```
Comment (pedrobaeza) [inline mass_mailing_partner/wizard/partner_merge.py]: Don't put empty lines inside a method

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**